lib/metasploit/framework/compiler/mingw.rb

```diff
@@ -43,90 +47,159 @@ def build_cmd(src)
           cmd << '-fno-ident '
           cmd << opt_level
 
-          if self.compile_options
-            cmd << self.compile_options
+          if compile_options
+            cmd << compile_options
           else
             link_options << '--image-base=0x0,'
             cmd << '-nostdlib '
           end
 
           link_options << '--no-seh'
-          link_options << ',-s' if self.strip_syms
-          link_options << ",-T#{self.link_script}" if self.link_script
+          link_options << ',-s' if strip_syms
+          link_options << ",-T #{link_script}" if link_script
 
           cmd << link_options
-
           cmd
         end
 
-        def cleanup_files
-          src_file = "#{self.file_name}.c"
-          exe_file = "#{self.file_name}.exe"
+        def compile_cpp(src, *additional_files)
+          within_temp(additional_files) do |dir, fnames|
+            file = Tempfile.create(['', '.cpp'], dir)
+            File.write(file, src)
+            if keep_src
+              dest = File.join(File.dirname(outfile), File.basename(file.path))
+              FileUtils.cp(file, dest)
+            end
+            fnames << file.path
+            cmd = build_cpp_files_cmd(fnames)
+            return exec(cmd), dest
```

Comment:
Slight comment but it is a bit odd this returns the results of `exec(cmd)` and `dest` whilst `compile_cpp_files` doesn't.

Reply:
nah, this is because `compile_cpp_files` doesn't create new source files, so there's nothing to return. the user already has this list of file becuase they passed it to method in the first place